144 Fix name capture for misc notifs + unit tests

taylorfinelli opened the pull request:
# What type of PR is this? (Check all that apply)

- [ ] ✨**feature**: Introduces completely new code or new features.
- [x] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [x] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

- Changed jest config file so that more components are tested
- Unit tests for `FriendRequestNotif` and `NudgeNotif` components
- Changed props for `FriendRequestNotif` and `NudgeNotif`, actually uses Notification type from Prisma DB
- Modified `getMiscNotifsFromUserId` -> `getMiscNotifsWithSenderUsernameFromUserId`
  - Now user the sender ID to get the sender username, returns with notification information

# Tests

How was this tested?

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

Explanation:
![Screenshot 2024-03-27 at 1 13 27 PM](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/e6e718c1-632b-4cb1-9a59-000d9acfaf35)

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

# Screenshots
https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/85365584/4e07309e-28ed-42f7-94e7-e188aa7e5709


# Additional notes
If you test locally, make sure to set the global context.

thienguen commented:
## 1

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/94078395/9e3c32b1-09e8-48e8-8565-2a5b87669be8)

Error here, also, need to run prettier to format this.

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/94078395/f110bd8a-82f9-476d-b7a1-9ff9903ad4bd)

You extract from id? as `number[]` I assume.


## 2

first run, after seeding the database, there are no notifications, it's empty, due to there being no notifications in the `notifications.json` having the receiver as `9` match with the mock user given through the global context

will you be updating it?

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/94078395/ed07f6a6-321d-4269-987e-96a3db2afefb)

After modifying it in the studio, I do see the notifications.

## 3

Also, just noticed now

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/94078395/835c2872-a49a-41a2-9765-091ef3987909)

It's colliding with the camera, I do see this becoming an issue.

thienguen commented:
![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/94078395/77f42439-cfe3-4d6e-a428-cafc4153f711)

they do be passing, very nice!

taylorfinelli commented:
# New changes

- Modified unit test snapshots since they were not passing (prettier reformatted the classnames so I just deleted the snapshots and re-ran tests)
- Refactored getMiscNotifsWithSenderUsernameFromUserId as per Thien's suggestions
- Modified TimeAgo unit tests so that snapshot will never be outdated
- Hid top nav bar
- Changed user data in global context so that it doesn't need to be changed to test functionality for this PR

# Future changes

- Will fix the spacing problem referenced in **2** of Thien's above comment in a separate issue
- Will make more mock data for User ID #4 so that we don't have to keep changing the data in global context

thienguen commented on apps/expo/src/components/timeAgo/TimeAgo.test.tsx:
```diff
@@ -1,18 +1,35 @@
+ * Thus, the snapshot will now never be out of date.
+ */
+function getDateFourWeeksAgo(): Date {
+  const currentDate = new Date()
```
I have fixed this one in my pr, and it tested throughout the TimeAlgo and the utility function that goes with it. 

You might expect to see merge conflict on the TimeAlgo wise

thienguen commented on packages/api/src/router/notif.ts:
```diff
@@ -1,55 +1,53 @@
-        return []
-      }
+    if (chat == undefined) {
+      return []
```
This is great, one thing if I were to beware of is

![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/94078395/9398066d-04a7-452f-953a-011a7e18da73)

For when you do a query or a mutation and the call doesn't exist data which leads to an early return. You can simply error out here, which will not break our app, but rather inform us through our console (terminal) that the query gets back nothing, it also returns an empty array (or object {} to the client). 

This will help better debug experience in the future, as return an [] will just let the app go on.

A minor suggestion, not a mandatory one.

```
import { TRPCError } from '@trpc/server'
```

taylorfinelli replied:
I have another issue where I am making changes to these APIs. Can I just implement this change in that issue instead of this one?

thienguen replied:
You can do so, yes

thienguen reviewed (commented):
Leave comments, everything else works as intended
